sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2669,6 +2670,41 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         ctx.EXTENDED != null)
     }
 
+  private def cleanTableProperties(
+      properties: Map[String, String],
+      location: Option[String],
+      isOptionClause: Boolean,
+      ctx: ParserRuleContext): Map[String, String] = withOrigin(ctx) {
+    import org.apache.spark.sql.connector.catalog.TableCatalog._
+    val legacyOn = conf.getConf(SQLConf.LEGACY_PROPERTY_NON_RESERVED)
+    var pathDefined = false
+    properties.filter {
```

Comment:
can we update `cleanNamespaceProperties` to use filter as well?

Reply:
done